app/controllers/api/v0/analytics_controller.rb

```diff
@@ -48,17 +52,32 @@ def past_day
       private
 
       def get_authenticated_user!
-        api_token = ApiSecret.find_by(secret: params[:api_token])
-        not_authorized if api_token.blank?
-        user = api_token.user
+        user = if request.headers["HTTP_API_KEY"].blank?
```

Comment:
Hi Andy! Nice refactoring.

I have one additional note regarding headers and stuff: HTTP headers are in the form of dash separated (case insensitive) words. For example `Last-Modified` and `If-Modified-Since` are headers that help HTTP deal with caching. To follow the spec/pattern the API key header could be something simply called `Api-Key` instead of `HTTP_API_KEY`. 

So for example, if I call from the client with:

```shell
$ curl -H "Api-Key: abcdef" http://localhost:3000/api/analytics/historical?start=3
```

the server will see:

```ruby
(byebug) request.headers["api-key"]
"abcdef"
```

You can read more about HTTP headers on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers)

Reply:
Thanks for reviewing again! I think Rails adds an extra `HTTP_` to any custom headers, so I defaulted to `["HTTP_API_KEY"]`. I think I prefer that since it's a case insensitive solution.

I ran `curl -H "api-key: 123" localhost:3000/api/analytics/totals` and got these results:
![Screen Shot 2019-04-03 at 5 41 50 PM](https://user-images.githubusercontent.com/17884966/55515157-e06de800-5637-11e9-9036-f303f3e41637.png)

From the looks of it, any of the following would work:
```ruby
request.headers["api-key"]
request.headers["Api-key"]
request.headers["Api-KEy"]
request.headers["API-KEY"]
request.headers["HTTP_API_KEY"]
```

Since that's the case, I think as long as we have good documentation for the API and let people know to use the header `api-key` (case insensitive), this should work for now. :)

Reply:
Also truth be told I spent about an hour trying to figure out why my headers weren't working. Turns out the issue was because of how Rails interacts with Postman, the API tool I was using. I finally ran into a search result that showed me how to use `curl` properly, and by the end of it I was just _done_ with headers. 🙊

Reply:
> From the looks of it, any of the following would work

Yeah, it's a case insensitive string. The last two I totally didn't know about. My suggestion was because it'd also be easier to grep for "api-key" given the correspondence in the documentation but yeah, it works :)

> I finally ran into a search result that showed me how to use curl properly, and by the end of it I was just done with headers. 

Ah ah I feel you!